tests/ccl_test_power.py

```diff
@@ -0,0 +1,104 @@
+import numpy as np
+from numpy.testing import assert_raises, assert_warns, assert_no_warnings, \
+                          assert_, decorators, run_module_suite
+import pyccl as ccl
+import sys
+
+# Set up the cosmological parameters to be used in each of the models
+# Values that are the same for all 5 models
+Omega_c = 0.25
+Omega_b = 0.045
+N_nu_rel = 0.
+h = 0.7
+sigma8 = 0.83
+n_s = 0.96
+
+# Values that are different for the different models
+Omega_v_vals = np.array([0.7, 0.7, 0.7, 0.65, 0.75])
+w0_vals = np.array([-1.0, -0.9, -0.9, -0.9, -0.9])
+wa_vals = np.array([0.0, 0.0, 0.1, 0.1, 0.1])
+
+# List of supported transfer functions
+transfer_fns = ['boltzmann', 'bbks', 'boltzmann_camb', 'boltzmann_class', 
+                'camb', 'class', 'eisenstein_hu', 'emulator', 
+                'fitting_function', 'none']
+
+transfer_fns = ['boltzmann_class', 
+                'class', 'eisenstein_hu', #'emulator', 
+                'fitting_function',]
+
+def all_finite(vals):
+    """
+    Returns True if all elements are finite (i.e. not NaN or inf).
+    """
+    return np.alltrue( np.isfinite(vals) )
+
+
+def calc_power_spectrum(Omega_v, w0, wa, transfer_fn, matter_power, linear):
+    """
+    Calculate linear and nonlinear power spectrum for a given set of parameters 
+    and choices of transfer function and matter power spectrum.
+    """
+    k = np.logspace(-4., 0., 300)
```

Comment:
Perhaps extend k range to below/above extrapolation ranges, to make sure extrapolation doesn't result in errors.